File: 7-reverse-integer/reverse-integer.java
class Solution {
    public long rev(long x){
        long res = 0;
        while (x != 0){
            long last = x %10;
            res = (res *10) +last;
            x = x /10;
        }
        return res;
    }

    public int reverse(int x) {
        if (x > 0){
            long res = rev(x);
            if(res > Integer.MAX_VALUE){
                return 0;
            }
            return (int)res;
        }
        else {
            long y = (long)x;
            long res = rev(Math.abs(y));
            if (res > Integer.MAX_VALUE){
                return 0;
            }
            return (int)-res;
        }
    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution:


    def rev(self,x):
        res = 0
        while x != 0:
            last = x % 10
            res = (res*10)+last
            x = x//10
        return res

    def reverse(self, x: int) -> int:
            if x >0:
                res = self.rev(x)
                if res < 2147483647:
                    return res
                return 0
            else:
                x = x * (-1)
                res = -1 * self.rev(x) 
                if res > -2147483648:
                    return res
                else:
                    return 0 
            